packages/jellyfish-api-core/__tests__/category/masternode/getMasternodeBlocks.test.ts

```diff
@@ -0,0 +1,167 @@
+    await masterNodeProvider1.container.restart()
+
+    await masterNodeProvider1.container.generate(20)
+    const mintedBlockNumber = 20
+    await masterNodeProvider1.container.generate(mintedBlockNumber, address)
+
+    const identifier: MasternodeBlock = {
+      id
+    }
+    const depth = 1
+    const masternodeBlock = await masterNodeProvider1.rpc.masternode.getMasternodeBlocks(identifier, depth)
+    expect(Object.keys(masternodeBlock).length).toEqual(depth)
+  })
+
+  it('should mint blocks by different masternode correctly and the depth is correct', async () => {
+    const masterNode1Addr = await masterNodeProvider1.container.getNewAddress('', 'legacy')
+    const masterNode1Id = await masterNodeProvider1.container.call('createmasternode', [masterNode1Addr])
+    await masterNodeProvider1.container.generate(1)
+
+    await masterNodeProvider1.container.setDeFiConf([`masternode_operator=${masterNode1Addr}`])
+    await masterNodeProvider1.container.restart()
+    const masterNode1Identifier: MasternodeBlock = {
+      id: masterNode1Id
+    }
+
+    const masterNode2Addr = await masterNodeProvider2.container.getNewAddress('', 'legacy')
+    const masterNode2Id = await masterNodeProvider2.rpc.masternode.createMasternode(masterNode2Addr)
+    await masterNodeProvider2.container.generate(1)
+
+    await masterNodeProvider2.container.setDeFiConf([`masternode_operator=${masterNode2Addr}`])
+    await masterNodeProvider2.container.restart()
+    const masterNode2Identifier: MasternodeBlock = {
+      id: masterNode2Id
+    }
+
+    await masterNodeProvider1.container.generate(20)
+    await masterNodeProvider2.container.generate(20)
+
+    const numberOfBlockPerMasterNode = 10
+    for (let i = 0; i < numberOfBlockPerMasterNode; i++) {
+      await masterNodeProvider1.container.generate(1, masterNode1Addr)
+      await masterNodeProvider1.container.generate(1)
+      await masterNodeProvider2.container.generate(1, masterNode2Addr)
+      await masterNodeProvider2.container.generate(1)
+    }
+    const masternode1Block = await masterNodeProvider1.rpc.masternode.getMasternodeBlocks(masterNode1Identifier)
+    const masternode2Block = await masterNodeProvider2.rpc.masternode.getMasternodeBlocks(masterNode2Identifier)
+
+    expect(Object.keys(masternode1Block).length).toEqual(numberOfBlockPerMasterNode)
+    expect(Object.keys(masternode2Block).length).toEqual(numberOfBlockPerMasterNode)
+
+    const depth = 4
+    const masternode1BlockDepth = await masterNodeProvider1.rpc.masternode.getMasternodeBlocks(masterNode1Identifier, depth)
+    const masternode2BlockDepth = await masterNodeProvider2.rpc.masternode.getMasternodeBlocks(masterNode2Identifier, depth)
+
+    // currently the depth = 4 does not return the last 4 blocks by each masternode,
+    // instead, it only returns the blocks that are minted by masternode in the last 4 latest block globally,
+    // this may not be the expected behaviour of the rpc call
```

Comment:
if the RPC returns the correct number of blocks minted by each MN(i.e 2) within the depth of 4, then that should be the correct behavior. But the test is not accurate 100%. This is because of not waiting for 2 MNs to sync. when one MN mint blocks, and if we need to make sure that blocks are propagated to other nodes in the network(i.e other MN in this case), we have to do `tGroup.waitForSync()`. you can check its usage in the existing tests. Refactor the test and then check the results.

Reply:
yea i did try to use `tGroup.waitForSync()` however it causes some timeout issue

Reply:
https://github.com/DeFiCh/jellyfish/issues/915

Reply:
the RPC has the correct behavior for the alternative minting test.